merge dz6 to master

3RUN opened the pull request.

3RUN committed:
first upload

3RUN committed:
add dz6 to menu

3RUN committed:
added singleton and student entity

Commit:
update

added basic layouts, adapter, view holder + singleton to store students (in mutable map)

Commit:
update

- filter edit text added
- storing data in singleton now
- string name fixes in resoures + dz3

Commit:
update

- working logic

Commit:
fixes

- removed url from details (not listed in home work)
- centered all info in details
- moved url field to the top in edit activity

Commit:
updated

- added working search by student's name

Commit:
fixes

- moved all values, colors etc into resources

Commit:
fixes

- various fixes in previous dzs

tim-teacher commented on app/src/main/java/by/itacademy/pvt/dz6/Dz6StudentEditActivity.kt:
```diff
@@ -0,0 +1,147 @@
+            loadStudentDetails(student, nameText, ageText, urlText)
+
+            saveButton
+                .setOnClickListener {
```
лучше все же логику разделять внутри клика а не 2 клика делать, ну и в методы бы повыносить

3RUN replied:
исправил, спасибо за ревью!

Commit:
fixes

- merged two save button click listeners into one
- moved save button logic into a seperate method
